### DISCUSSION TITLE
Compatability between scipy.signal and python-control

### DISCUSSION THREAD

**Discussion by bavcol** [python-control/python-control#956] [General]
I was exploring the idea to migrate from MATLAB to python for signal processing (scipy.signal) and control (python-control).
It was quite a bummer when I realized each module has its own LTI representation and the two are not compatible.
This seems like a serious issue to me, because now I have to handle / convert the two conflicting LTI representations when I want to use both modules.
- Why does python-control use LTI representations that are incompatible to scipy.signal?
- Are there any plans to improve the compatability between scipy.signal and python-control?
---
FYI: I searched for similar discussions, but I could only dig up [this old comment](https://github.com/python-control/python-control/issues/48#issuecomment-84219603), where the question is raised as a side topic, but not further discussed.

**Comment by murrayrm** [comment]
Is there a specific example where you would see yourself shifting back and forth between scipy.signal and python-control?  That might help sort out if there a good way to make the systems more compatible.

One approach to making things more compatible would be to create a `to_scipy` method for all relevant python-control objects and then allow the python-control factory functions (`ct.ss`, `ct.tf`, etc) accept scipy.signal objects and convert them.

I think the main challenge with making the packages completely compatible is that python-control adds functionality (eg, signal names, interconnection functions, nonlinear systems) that scipy.signal seems unlikely to need.
